src/PHPUnit/Controller/AbstractControllerTestCase.php

```diff
@@ -380,8 +382,9 @@ public function triggerApplicationEvent($eventName)
      */
     public function assertModulesLoaded(array $modules)
     {
+        /** @var ModuleManager $moduleManager */
         $moduleManager = $this->getApplicationServiceLocator()->get('ModuleManager');
```

Comment:
```suggestion
        $moduleManager = $this->getApplicationServiceLocator()->get('ModuleManager');
        assert($moduleManager instanceof ModuleManager);
```

Try to use an assertion in cases like this - it has the same effect of telling your IDE and psalm what the type is expected to be, but with the added benefit of causing a runtime error when the type does not match with a clear assertion failure.

When the type should be verified as part of expectations for the subject under test, then PHPUnit's `self::assertInstanceOf()` method should be used.

Reply:
Good point.  Thanks